bql/planner/planner_test.go

```diff
@@ -566,6 +566,26 @@ func TestPlannerQuery(t *testing.T) {
 			nbs:  2,
 			nrws: 4,
 		},
+		{
+			q: `SELECT ?s_id, ?height
+					FROM ?test
+					WHERE {
+						?s ID ?s_id "height_cm"@[] ?height
+					}
+					HAVING ?s_id = "alice"^^type:text;`,
+			nbs:  2,
+			nrws: 1,
+		},
+		{
+			q: `SELECT ?p_id, ?o
+					FROM ?test
+					WHERE {
+						/u<peter> ?p ID ?p_id ?o
+					}
+					HAVING ?p_id < "parent_of"^^type:text;`,
+			nbs:  2,
+			nrws: 4,
+		},
```

Comment:
Should we add tests comparing with a literal of a "incorrect" type?
Maybe also add a test with a node ID and a node literal?

Reply:
For now I added the tests for these two cases inside `expression_test.go`, since there we already had a structure to verify expected errors.
Should I add a similar structure inside `planner_test.go` too (like an `err` entry below `nrws` for `testTable`, inside `TestPlannerQuery`) and also repeat these tests there?

Reply:
I have just added a similar structure for checking expected errors inside `planner_test.go` too, and added additional tests there and inside `expression_test.go` for code coverage.

Reply:
I have just added one more check for literal type coherence when comparing literals inside `expression.go`, along with some tests for that too.